prometheus/desc.go

```diff
@@ -57,6 +57,8 @@ type Desc struct {
 	// must be unique among all registered descriptors and can therefore be
 	// used as an identifier of the descriptor.
 	id uint64
+	// compatID is similar to id, but is a hash of all the relevant names escaped with underscores.
+	compatID uint64
```

Comment:
When I first saw this variable I thought it was a typo for compactID, but I see you're are using compact is other places as well.

What does compat means? 😅

Reply:
And still related to this variable, have we thought about using id to store the hash of the already normalized descriptor?

Reply:
"compatible." as in, "compatible with legacy systems." I need to pick better names for sure

Reply:
> have we thought about using id to store the hash of the already normalized descriptor?

do you mean only storing the normalized descriptor? We need both in case the user switches the mode to UTF-8 comparison with no legacy compatibility. Or do you mean something else?

Reply:
Yes, I meant just storing the normalized descriptor. Good point that's it's possible to change the validation in-flight!

Reply:
Not sure why someone would like to do that though 🤔

Reply:
for the record (as discussed in slack), it's because metrics are often created in `init()` functions before the user has a chance to actually configure the correct escaping scheme. So we have to be ready for both scenarios.